apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/Rates/Filter.jsx

```diff
@@ -0,0 +1,80 @@
+// @flow
+import React, { useState } from "react";
+import { Trans } from "react-i18next";
+import Box from "~/renderer/components/Box";
+import styled from "styled-components";
+import { FILTER } from "../utils";
+import Button from "~/renderer/components/Button";
+import { Icons } from "@ledgerhq/react-ui";
+
+type Props = {
+  onClick: () => void,
+};
+
+const Container: ThemedComponent<{}> = styled(Box).attrs(() => ({}))`
+  flex-direction: row;
+  margin: 10px 0;
+`;
+
+export const Btn: ThemedComponent<{}> = styled(Button).attrs((p) => {
+  return {
+    key: p.key,
+    primary: p.selected,
+    outline: p.selected,
+    onClick: () => p.updateFilter(p.type),
+    iconPosition: "left",
+  };
+})`
+  border-radius: 500px;
+  margin-right: 12px;
+  padding: 0 12px;
+  height: 32px;
+  color: ${(p) => (p.selected ? p.theme.colors.neutral.c00 : p.theme.colors.neutral.c100)};
+  background-color: ${(p) =>
+    p.selected ? p.theme.colors.primary.c80 : p.theme.colors.neutral.c30};
+  &:active,
+  &:hover {
+    color: ${(p) => (p.selected ? p.theme.colors.primary.c80 : p.theme.colors.neutral.c100)};
+  }
+`;
+
+export default function ProviderRate({ onClick }: Props) {
+  const [filter, setFilter] = useState([]);
+
+  const updateFilter = (type) => {
```

Comment:
use a `useCallback` will avoid the `<Btn>` to always rerender (as the updateFilter prop will always change ref)

Reply:
Done!